### PR TITLE
fix(vector): move minReadySeconds to global scope

### PR DESCRIPTION
**Pull request by rbakhtaraev**
In this case, we must use the global (root) context for access to `.Values.minReadySeconds` variable.
Without it, Helm try to find it in `.Values.updateStrategy` scope.

closes: #306

### CODE REVIEW

**Comment by neuronull** [comment]
I validated this locally, thanks for dropping a fix!

**Comment by neuronull** [comment]
The chart version just needs to be upgraded here: https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/Chart.yaml#L3

**Comment by neuronull** [comment]
Ah, the changelog doesn't need to be modified in the individual PRs, we update it with a script as part of the release process: https://github.com/vectordotdev/helm-charts#releasing

**Review by spencergilbert — APPROVED** [review]

**Comment by kaden-l-nelson** [comment]
@spencergilbert What're your thoughts on taking this PR instead? https://github.com/vectordotdev/helm-charts/pull/308/files

#308 makes it so that min ready seconds can be configured in the absence of an update strategy.

**Comment by skygrammas** [comment]
@kaden-l-nelson makes a good point. In the absence of an update strategy value, would the [`{{- with .Values.updateStrategy }}` block](https://github.com/vectordotdev/helm-charts/blob/b2e52a3d81570faa52659c2c894255aa422b66da/charts/vector/templates/daemonset.yaml#L14) not be generated? Resulting in `minReadySeconds` not being templated despite it being reference in the root context?

**Comment by spencergilbert** [comment]
> @kaden-l-nelson makes a good point. In the absence of an update strategy value, would the [`{{- with .Values.updateStrategy }}` block](https://github.com/vectordotdev/helm-charts/blob/b2e52a3d81570faa52659c2c894255aa422b66da/charts/vector/templates/daemonset.yaml#L14) not be generated? Resulting in `minReadySeconds` not being templated despite it being reference in the root context with this proposed change?

That's correct - it does look like the `with` line isn't in the right place. There doesn't need to be any connection between the `updateStrategy` and the `minReadySeconds` values so they should be templated separately.

cc @neuronull

File: charts/vector/templates/daemonset.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ spec:
     matchLabels:
       {{- include "vector.selectorLabels" . | nindent 6 }}
   {{- with .Values.updateStrategy }}
```
Comment (skygrammas) [inline charts/vector/templates/daemonset.yaml:14]:
```suggestion
   minReadySeconds: {{ $.Values.minReadySeconds }}
   {{- with .Values.updateStrategy }}
```

@rbakhtaraev would you mind swapping these lines in this proposal?

File: charts/vector/templates/daemonset.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ spec:
...
-  minReadySeconds: {{ .Values.minReadySeconds }}
+  minReadySeconds: {{ $.Values.minReadySeconds }}
```
Comment (skygrammas) [inline charts/vector/templates/daemonset.yaml:15]:
```suggestion
```

File: charts/vector/templates/daemonset.yaml (hunk 1 above)
Comment (spencergilbert) [reply charts/vector/templates/daemonset.yaml:14]: If moved outside of the `with` block, the `$` should be dropped as well, I beleive.

Comment (kaden-l-nelson) [reply charts/vector/templates/daemonset.yaml:14]: https://github.com/vectordotdev/helm-charts/pull/308 This PR already has the correct changes.

Comment (skygrammas) [reply charts/vector/templates/daemonset.yaml:14]: This is true. @spencergilbert how do you feel about reviewing the other PR instead? Or would you prefer moving forward on this proposal?

Comment (rbakhtaraev) [reply charts/vector/templates/daemonset.yaml:15]: yep, fixed it

Comment (skygrammas) [reply charts/vector/templates/daemonset.yaml:15]: awesome, thanks.

**Comment by spencergilbert** [comment]
@rbakhtaraev looks like this needs a `helm-docs` run to update the badges.

**Pull request merged**